linux_new/Jenkinsfile

```diff
@@ -0,0 +1,887 @@
+                    changelogversion = "${Release}.${Version}.${TemurinVersion}+${Build}"
+                }
+            }
+
+            if (PARCH == 'arm') {
+                // If JDK8 ARM32, Then Need To Deduce Correct Upstream
+                upstreamversionARM32 = upstreamversionARM32.replaceFirst("^jdk", "").replaceAll(/-(jre|jdk)$/, "")
+                // DONE
+            }
+
+            // Figure Out Template name
+            templatebase = "./linux_new/${PTYPE}/${DistArrayElement}/src/main/packaging/${PRODUCT}/${Release}/${PRODUCT}-${Release}-${PTYPE}.template.j2"
+            // Output File Name Only ( defaults to outputting in same location as template )
+            outputfile = "${PRODUCT}-${Release}-${PTYPE}.spec"
+
+            // Check If Template Exists
+            if (!fileExists(templatebase)) {
+                error("Template File Not Found At : ${templatebase}")
+            }
+
+            def speccmd = "python3 linux_new/generate_spec.py \"${templatebase}\" \"${packagever}\" \"${packagearchRhel}\" \"${PKURL}\" \"${PCSUM}\" \"${PFILE}\" \"${outputfile}\" \"${currentDateRHEL}\" \"${PackageReleaseVersion}\" \"${upstreamversion}\" \"${changelogversion}\" \"${upstreamversionARM32}\""
+            def genresult = sh(script: speccmd, returnStatus: true)
+            // End Of RHEL SUSE CODE
+        }
+    }
+}
+
+// Archive & store the generated files, for future reference
+
+ArchiveFileName = "Package_Bld_Src_${distro}_${params.TAG}_${arch}_${env.BUILD_NUMBER}.tar.gz"
+echo "ArchiveFileName = ${ArchiveFileName}"
+sh "find ./linux_new -type f \\( -name '*.spec' -o -name 'control' -o -name 'changelog' -o -name 'rules' -o -name 'APKBUILD' \\) | tar -czf ./${ArchiveFileName} -T -"
+
+// Publish the tarball
+archiveArtifacts artifacts: "${ArchiveFileName}", allowEmptyArchive: false
+
+// Stash the generated build files for use throughout the rest of the process
+dir('linux_new') {
+    stash name: 'installercode', includes: '**'
+}
+}
+}
+}
+// Generate Spec File Stage - End
+// Build And Archive Packages Stage - Start
+stage('Build & Archive Package') {
+    when {
+        expression { return params.DRY_RUN == false }
+    }
+    steps {
+        script {
+            echo "Entering Stage : Build & Archive Package"
+            DISTS_TO_BUILD.each { DistArrayElement ->
+            def nodeLabel = ''
+            if (DistArrayElement == 'alpine') {
+                nodeLabel = PKGBUILDLABELAPK
+            } else if (DistArrayElement == 'debian') {
+                nodeLabel = PKGBUILDLABELDEB
+            } else if (DistArrayElement in ['rhel', 'suse']) {
+                nodeLabel = PKGBUILDLABELRHEL
+            } else {
+                error "Unknown DistArrayElement: ${DistArrayElement}"
+            }
+
+            node(label: nodeLabel) { // Assign node dynamically based on distribution type
+
+            // Docker --mount option requires BuildKit
```

Comment:
Not an issue for the purposes of this PR, but do we know if the use of docker here would work for systems running `podman` too?

Reply:
I'll add a note to phase 2 to try this out, it was something I'd considered, but time was the enemy.